source/Plugins.cpp

```diff
@@ -65,10 +66,36 @@ const Plugin *Plugins::Load(const string &path)
 	size_t pos = path.rfind('/', path.length() - 2) + 1;
 	string name = path.substr(pos, path.length() - 1 - pos);
 
-	auto *plugin = plugins.Get(name);
-	plugin->name = std::move(name);
+	auto * plugin = plugins.Get(name);
+
+	string pluginFile = path + "/plugin.txt";
+	bool hasPluginFile = Files::Exists(pluginFile);
+
+	// Loads plugin metadata from plugin.txt.
+	DataFile file(path + "/plugin.txt");
+	for(const DataNode & child : file)
```

Comment:
```suggestion
	for(const DataNode &child : file)
```

Reply:
Normally the first layer would be called node, right?
Progression like this: node, child, grand

Reply:
yup